language/tools/move-mv-llvm-compiler/src/stackless/llvm.rs

```diff
@@ -0,0 +1,444 @@
+//! LLVM wrappers.
+//!
+//! The stackless code generator accesses llvm only through this mod.
+//!
+//! It:
+//!
+//! - Runs dtors
+//! - Encapsulates unsafety, though making LLVM fully memsafe is hard.
+//! - Hides weirdly mutable array pointers.
+//! - Provides high-level instruction builders compatible with the stackless bytecode model.
+
+use llvm_extra_sys::*;
+use llvm_sys::core::*;
+use llvm_sys::prelude::*;
+use llvm_sys::target::*;
+use llvm_sys::target_machine::*;
+
+use crate::cstr::SafeCStr;
+
+use std::ptr;
+use std::ffi::{CStr, CString};
+
+pub use llvm_extra_sys::AttributeKind;
+pub use llvm_sys::LLVMIntPredicate;
+
+pub fn initialize_bpf() {
+    unsafe {
+        LLVMInitializeBPFTargetInfo();
+        LLVMInitializeBPFTarget();
+        LLVMInitializeBPFTargetMC();
+        LLVMInitializeBPFAsmPrinter();
```

Comment:
It probably does not matter today, but we may soon want to start using the new SBF back-end here (LLVMInitializeSBF*) instead.

Reply:
Yes. At the moment I am using the old BPF target definitions, not sure how ready SBF2 is, but I expect to switch to SBF at some point since if the move backend is ever finished it will be well after SBF2 is deployed.

Reply:
As far as I know, the default back-end today is the SBF back-end. That is, I believe all of our ordinary production scripts should be using `sbf-solana-solana`. So at the moment, target `sbf-solana-solana` generates roughly the same code as target `bpfel-unknown-unknown` with `+solana`. The ELF `e_machine` values differ, of course (`EM_BPF` vs `EM_SBF`).

This new back-end is part of the overall transition away from BPF to SBF in our tools, scripts, etc (https://github.com/solana-labs/solana/issues/19113). If by SBF2 you mean SBFv2 (https://github.com/solana-labs/solana/issues/20323), some of that can be considered orthogonal work (enhancing/modifying the ISA, etc) to happen later.

Reply:
Thanks for the clarifications. I wasn't not sure what the distinction between sbfv1 and sbfv2 I had seen in the code were, nor that the tooling had already switched to the sbf target.

I filed an issue https://github.com/solana-labs/move/issues/33